### ISSUE TITLE
chezmoi fails with userid lookup error

### ISSUE THREAD

**Issue by marascio** [twpayne/chezmoi#905]
## Describe the bug

Running `chezmoi diff` or any other command that performs work (`help` doesn't fail for example) results in the following error: 

```
$ chezmoi diff
chezmoi: user: lookup userid 10001: input/output error
```
## To reproduce

The only special consideration of our environment is that users and groups are defined in LDAP and are not in `/etc/passwd` and `/etc/group`

## Expected behavior

I would expect it to provide the diff of the config to what is currently installed.

## Output of command with `--verbose`

The output does not change, see below.

```
$ bin/chezmoi diff --verbose
chezmoi: user: lookup userid 10001: input/output error
```

## Output of `chezmoi doctor`

```
$ chezmoi doctor
     ok: version 1.8.6, commit cfa3fd2674bc7cd74cbf506f52ebe15723365133, built at 2020-09-28T15:41:30Z, built by goreleaser
     ok: runtime.GOOS linux, runtime.GOARCH amd64
     ok: /home/REDACTED_PATH_TO_DOTFILES (source directory, perm 700)
     ok: /home/REDACTED_HOME (destination directory, perm 755)
     ok: /home/REDACTED_HOME/.config/chezmoi/chezmoi.toml (configuration file)
     ok: /bin/bash (shell)
     ok: /usr/bin/vi (editor)
     ok: /usr/bin/vimdiff (merge command)
     ok: /usr/bin/git (source VCS command, version 2.25.1)
     ok: /usr/bin/gpg (GnuPG, version 2.2.19)
warning: op (1Password CLI, not found)
warning: bw (Bitwarden CLI, not found)
warning: gopass (gopass CLI, not found)
warning: keepassxc-cli (KeePassXC CLI, not found)
warning: lpass (LastPass CLI, not found)
warning: pass (pass CLI, not found)
warning: vault (Vault CLI, not found)
```

**Comment by twpayne** [comment]
chezmoi uses the Go standard library's [`user.Current()`](https://pkg.go.dev/os/user#Current) function to look up your username. It looks like this is returning an error in your case.

Please can you create a file called `user.go` containing:

```go
package main

import (
	"fmt"
	"os/user"
)

func main() {
	fmt.Println(user.Current())
}
```

and then run it with

```
go run user.go
```

and report the output here.

For reference, on my macOS machine, it outputs

```
&{501 20 twp Tom Payne /Users/twp} <nil>
```

**Comment by twpayne** [comment]
Did this break recently, i.e. did it used to work with chezmoi 1.8.5? If so, it's almost certainly due to #893 which switched the libc from glibc to musl.

**Comment by marascio** [comment]
Sorry it took me a few days to get back to this. See below (I've redacted my user name and full name):

```
$ go version
go version go1.15.2 linux/amd64
$ go run user.go
&{10001 10000 USER NAME /home/USER} <nil>
$ which chezmoi
/home/USER/bin/chezmoi
$ chezmoi --version
chezmoi version 1.8.6, commit cfa3fd2674bc7cd74cbf506f52ebe15723365133, built at 2020-09-28T15:41:30Z, built by goreleaser
$ chezmoi diff
chezmoi: user: lookup userid 10001: input/output error
```

**Comment by marascio** [comment]
> Did this break recently, i.e. did it used to work with chezmoi 1.8.5? If so, it's almost certainly due to #893 which switched the libc from glibc to musl.

In the past I have only used chezmoi on my workstations and laptops. I'm trying to use it now on various remote machines and servers. The key differences between my own workstations/laptops and the systems where I am seeing this error are:

1. My workstations/laptops are either OSX or Arch Linux while the systems that I am seeing this error on are Ubuntu 20.04.
2. The systems where I am seeing these errors use LDAP for users and groups. Users are not defined locally at all and don't exist in `/etc/passwd` or `/etc/group`.

I suspect #2 above is the key difference, as I doubt chezmoi has any real issue running on Ubuntu.

**Comment by marascio** [comment]
@twpayne I installed chemzoi 1.8.5 and it works. I don't see the error any longer so it appears your hunch was correct and it was #893 and the switch to musl.

**Comment by twpayne** [comment]
Thank you very much for your detailed investigation @marascio!

I propose to fix this in the following way:
* Keep musl for the the statically-linked single-binary releases.
* Revert to glibc for the packaged binaries (.rpm, .deb, etc.).

**Comment by twpayne** [comment]
I've now released version 1.8.7 which includes the fix. Please re-open this issue if this does not solve the problem.
